src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1123,6 +1273,54 @@ SingleResult ToResult(TimeSeriesOperation.AppendOperation element)
             }
         }
 
+
+        private static readonly byte[] TimedCounterPrefixBuffer = Encoding.UTF8.GetBytes(TimedCounterPrefix);
+        private static readonly byte[] TimedCounterPositivePrefixBuffer = Encoding.UTF8.GetBytes(TimedCounterPositivePrefix);
+        private const string TimedCounterPrefix = "TC:";
+        private const string TimedCounterPositivePrefix = TimedCounterPrefix + "INC-";
+        private const string TimedCounterNegativePrefix = TimedCounterPrefix + "DEC-";
+        public string IncrementTimestamp(
+            DocumentsOperationContext context,
+            string documentId,
+            string collection,
+            string name,
+            IEnumerable<TimeSeriesOperation.IncrementOperation> toIncrement)
+        {
+#if DEBUG
+            DateTime prevTime = DateTime.MinValue;
+#endif
+
+            var holder = new SingleResult();
+
+            return AppendTimestamp(context, documentId, collection, name, toIncrement.Select(ToResult));
+
+
+            SingleResult ToResult(TimeSeriesOperation.IncrementOperation element)
+            {
+#if DEBUG
+                if (element.Timestamp <= prevTime)
+                {
+                    throw new InvalidDataException(
+                        $"The order of increment operations must be sequential, but got previous operation {prevTime} and the current: {element.Timestamp}");
+                }
+
+                prevTime = element.Timestamp;
+#endif
+                var sign = Math.Sign(element.Values[0]);
+                for (int i = 1; i < element.Values.Length; i++)
+                {
+                    if (sign != Math.Sign(element.Values[i]))
+                        throw new InvalidDataException("Cannot mix increment & decrement operations in a single call. But got: " + string.Join(", ", element.Values));
+                }
+
+                holder.Values = element.Values;
+                holder.Tag = context.GetLazyString((element.Values[0] > 0 ? TimedCounterPositivePrefix : TimedCounterNegativePrefix) + _documentDatabase.DbBase64Id);
```

Comment:
I would prefer `0` to be positive?... or even better to skip the `0`s ?

Reply:
Done